### PR TITLE
 BaSP-MR-80: post and put methos in Trainer

### PR DESCRIPTION
**Pull request by Emanuel-Lamberti**
**Description**
My [Trello's ticket](https://trello.com/c/3CMwFXFf/80-trainer-add-edit-form) describes that i had to should do conditional rendering based on a momentary state. 
To do this, develop methods to add and edit (by id) the coaches that will later be added to the table.

_I used the global Modals on my table and made a little change on it (with consent of the creator)._

**Changes Made**
On Trainers component

- -index.js and its css added.
- -created Table component that contains an index.js .
- -created a table.module.css which i used to give styles to the table.
- -add button on table and gave them styles as well.

**Checklist**

- [ ]  -Create add/edit form including necessary fields
- [ ]  -On edit functionality should show previous info
- [ ]  -Success message for confirmation edit and add actions
- [ ]  -Error message for edit and create actions
- [ ]  -Add "add new admin" button

### CODE REVIEW

File: src/Components/Trainers/FormEditTrainers/FormEditTrainer.jsx [hunk 1]
```diff
@@ -0,0 +1,261 @@
+import React, { useState } from 'react';
+import styles from './form.module.css';
+import ModalConfirm from '../../Modals/ModalConfirm';
+import ModalSuccess from '../../Modals/ModalSuccess';
+
+// eslint-disable-next-line no-unused-vars
```
Comment (FrancoLelli) [inline src/Components/Trainers/FormEditTrainers/FormEditTrainer.jsx:6]:
```suggestion

```

File: src/Components/Trainers/FormTrainers/index.jsx [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+    if (trainer.salary.trim() === '') {
+      newErrors.city = '*City is required';
+      isValid = false;
+    }
+
+    setErrors(newErrors);
+    return isValid;
+  };
+
+  // eslint-disable-next-line no-unused-vars
```
Comment (FrancoLelli) [inline src/Components/Trainers/FormTrainers/index.jsx:81]:
```suggestion
 
```

File: src/Components/Trainers/TableTrainers/index.jsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
+import React, { useState } from 'react';
+import styles from './table.module.css';
+import FormEdit from '../FormEditTrainers/FormEditTrainer';
+
+// eslint-disable-next-line no-unused-vars
```
Comment (FrancoLelli) [inline src/Components/Trainers/TableTrainers/index.jsx:5]:
```suggestion
```

**Review by FrancoLelli — CHANGES_REQUESTED** [review]
Ema, nice Job, but i think you should delete comments!!

**Review by FrancoLelli — APPROVED** [review]

**Comment by flordimonaco** [comment]
> Ema, nice Job, but i think you should delete comments!!

I agree with fran!

**Review by CristianLotorto — CHANGES_REQUESTED** [review]
good code Ema, but you pushed .gitignore, package-lock.json and package.json :/

**Review by CristianLotorto — APPROVED** [review]

**Review by NicoCagnina — CHANGES_REQUESTED** [review]

**Comment by Emanuel-Lamberti** [comment]
Comments in the files have already been removed, and conflicts resolved.

**Review by NicoCagnina — APPROVED** [review]

**Pull request merged**